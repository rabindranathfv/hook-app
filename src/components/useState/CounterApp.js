import React, { useState } from 'react';
import './counter.css';

export const CounterApp = () => {

    const initialState = { counter: 0, counter2: 20 };
    const [state, setState] = useState(initialState);

    const { counter, counter2 } = state;

    const add = () => {
        setState({
            ...state,
            counter: counter + 1
        })
    }

    const subs = () => {
        setState({
            ...state,
            counter: counter - 1
        })
    }

    const reset = () => {
        setState({
            ...state,
            counter: 0
        })
    }

    return (
        <div>
            <h1> Counter1 { counter } </h1>
            <h1> Counter2 { counter2 } </h1>
            <hr />

            <button type="button" class="btn btn-outline-primary" onClick={add}> +1</button>
            <button type="button" class="btn btn-outline-primary" onClick={subs}> -1</button>
            <button type="button" class="btn btn-outline-primary" onClick={reset}> reset</button>
            <button type="button" class="btn btn-outline-primary"> +1</button>
        </div>
    )
}
