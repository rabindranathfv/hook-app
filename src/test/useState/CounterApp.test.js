import React from 'react';
import { shallow } from 'enzyme';
import { CounterApp } from '../../components/useState/CounterApp';

describe('CounterApp::', () => {

    let wrapper = shallow(<CounterApp />);

    beforeEach( ()=> {
        wrapper = shallow(<CounterApp />);
    })
    
    it('should return default values CounterApp', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render with default values in all counters', () => {
        const counterText = wrapper.find('h1').at(0).text().trim();
        const counterText2 = wrapper.find('h1').at(1).text().trim();

        expect(counterText).toBe('Counter1 0');
        expect(counterText2).toBe('Counter2 20');
    });

    it('Should increment counter', () => {
        wrapper.find('button').at(0).simulate('click');
        const counterText = wrapper.find('h1').at(0).text().trim();
        
        expect(counterText).toBe('Counter1 1');
    });

    it('Should decrement counter', () => {
        wrapper.find('button').at(1).simulate('click');
        const counterText = wrapper.find('h1').at(0).text().trim();
        
        expect(counterText).toBe('Counter1 -1');
    });

    it('Should decrement reset', () => {
        wrapper.find('button').at(2).simulate('click');
        const counterText = wrapper.find('h1').at(0).text().trim();
        
        expect(counterText).toBe('Counter1 0');
    });
})
