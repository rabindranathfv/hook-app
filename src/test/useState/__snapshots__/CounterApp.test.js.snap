// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterApp:: should return default values CounterApp 1`] = `
<div>
  <h1>
     Counter1 
    0
     
  </h1>
  <h1>
     Counter2 
    20
     
  </h1>
  <hr />
  <button
    class="btn btn-outline-primary"
    onClick={[Function]}
    type="button"
  >
     +1
  </button>
  <button
    class="btn btn-outline-primary"
    onClick={[Function]}
    type="button"
  >
     -1
  </button>
  <button
    class="btn btn-outline-primary"
    onClick={[Function]}
    type="button"
  >
     reset
  </button>
  <button
    class="btn btn-outline-primary"
    type="button"
  >
     +1
  </button>
</div>
`;
